### PR TITLE
Departments services

### PR DESCRIPTION
**Pull request by AnshulShirude**
## Description/Problem

Closes [Departments service/controller](copy_trello_card_link_here)

Implement Departments service and controller

## Solution

Added in the departments controller/services along with the tests for it.

## Dependencies

[X] This PR adds new dependencies

## Testing
Tested manually and wrote tests

### CODE REVIEW

File: apps/server/src/departments/departments.controller.spec.ts [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
 
     controller = module.get<DepartmentsController>(DepartmentsController);
+    departmentsService = module.get<DepartmentsService>(DepartmentsService);
   });
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
   });
+
+  describe('findAll', () => {
```
Comment (elvincheng3) [inline apps/server/src/departments/departments.controller.spec.ts:25]: can we test the other routes as well?

File: apps/server/src/departments/departments.service.spec.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { DepartmentsService } from './departments.service';
+import { PrismaService } from '../prisma/prisma.service';
 
 describe('DepartmentsService', () => {
   let service: DepartmentsService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [DepartmentsService],
```
Comment (elvincheng3) [inline apps/server/src/departments/departments.service.spec.ts:9]: can we also get some tests here for the departments service? can refer to the form instances pr for some reference

File: apps/server/src/employees/employees.service.ts [hunk 3]
```diff
@@ -22,11 +22,15 @@ export class EmployeesService {
         positionId: createEmployeeDto.positionId,
         pswdHash: await bcrypt.hash(
           createEmployeeDto.password,
-          process.env.SALT_ROUNDS || 10,
+          await bcrypt.genSalt(),
```
Comment (elvincheng3) [inline apps/server/src/employees/employees.service.ts:25]: does this take in a param for number of rounds?

Comment (AnshulShirude) [reply apps/server/src/employees/employees.service.ts:25]: yes, I think I can add in 10 as a param in there

File: apps/server/src/departments/departments.controller.spec.ts (hunk 1 above)
Comment (AnshulShirude) [reply apps/server/src/departments/departments.controller.spec.ts:25]: Made a new ticket. Will get to the testing later in the week hopefully.

File: apps/server/src/departments/departments.service.spec.ts (hunk 2 above)
Comment (AnshulShirude) [reply apps/server/src/departments/departments.service.spec.ts:9]: Made a new ticket. Will get to the testing later in the week hopefully.

File: apps/server/src/departments/departments.controller.spec.ts (hunk 1 above)
Comment (elvincheng3) [reply apps/server/src/departments/departments.controller.spec.ts:25]: https://trello.com/c/1IraTXeL/46-departments-controller-tests

File: apps/server/src/departments/departments.service.spec.ts (hunk 2 above)
Comment (elvincheng3) [reply apps/server/src/departments/departments.service.spec.ts:9]: https://trello.com/c/rQ3mXVoA/45-departments-service-tests

File: apps/server/src/employees/employees.service.ts [hunk 4]
```diff
@@ -22,7 +22,7 @@ export class EmployeesService {
         positionId: createEmployeeDto.positionId,
         pswdHash: await bcrypt.hash(
           createEmployeeDto.password,
-          process.env.SALT_ROUNDS || 10,
+          await bcrypt.genSalt(10),
```
Comment (elvincheng3) [inline apps/server/src/employees/employees.service.ts:25]: can we use the env variable with `10` as the fallback?

**Review by elvincheng3 — COMMENTED** [review]
small nit, but nice work!

File: apps/server/src/employees/employees.service.ts (hunk 4 above)
Comment (AnshulShirude) [reply apps/server/src/employees/employees.service.ts:25]: For that, we will need to use ConfigService and that will add a dependency to the folder. Should I try that?

Comment (elvincheng3) [reply apps/server/src/employees/employees.service.ts:30]: yep, that makes sense

**Comment by elvincheng3** [comment]
[Departments service/controller](https://trello.com/c/atfQrl1A/34-departments-service-controller)

File: apps/server/src/departments/departments.controller.spec.ts [hunk 5]
```diff
@@ -1,20 +1,164 @@
...
+          // Handle other IDs if needed.
+        });
+
+      await expect(
+        controller.findOne(invalidDepartmentId),
+      ).rejects.toThrowError(NotFoundException);
+    });
+  });
+
+  describe('update', () => {});
```
Comment (elvincheng3) [inline apps/server/src/departments/departments.controller.spec.ts:134]: tests tbd
https://trello.com/c/pMztE6Dd/50-add-controller-test-for-updating-department

**Review by elvincheng3 — APPROVED** [review]
🚀

**Comment by elvincheng3** [comment]
[Departments controller tests](https://trello.com/c/1IraTXeL/46-departments-controller-tests)

**Comment by elvincheng3** [comment]
[Departments service tests](https://trello.com/c/rQ3mXVoA/45-departments-service-tests)

**Pull request merged**